### PR TITLE
added support for Visual Studio Code Remote Containers, added

### PR DESCRIPTION
**Pull request by chafey**

### CODE REVIEW

**Comment by chafey** [comment]
This PR includes support for Visual Studio Code with Remote Containers extension.  It also includes a script to create a shell in a docker container that has EMSCRIPTEN so you can build there vs installing EMSCRIPTEN locally

**Comment by aous72** [comment]
I think this is very useful addition.
I will integrate it with the code in a couple of days.  There are only small modification that I wish to add.  I will also modify the README.md file.

Thank you Chris.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    && apt-get -y install git iproute2 procps lsb-release \
+    #
+    # Install C++ tools
+    && apt-get -y install build-essential cppcheck valgrind \
+    #
+    ## Install newer version of CMake (OpenJPH requires 3.10+, debian has older version)
+    && apt-get install wget \
+    && wget https://github.com/Kitware/CMake/releases/download/v3.15.2/cmake-3.15.2.tar.gz \
+    && tar -zxvf cmake-3.15.2.tar.gz \
+    && (cd /cmake-3.15.2; ./bootstrap; make -j12 install)\
```
Comment (aous72) [inline .devcontainer/Dockerfile]: -j12 is rather aggressive; you must have a machine with 12 or more cores, and a fast hard drive.
I will reduce it to a more modest -j2.

File: subprojects/js/emscripten-docker.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+docker run -it -v "$(cd ../.. && pwd)":/src trzeci/emscripten /bin/bash
```
Comment (aous72) [inline subprojects/js/emscripten-docker.sh]: I want to add
--user $(id -u):$(id -g)
to make the files generated have the user id and group.

**Comment by aous72** [comment]
This is all great. It has been integrated into the development branch.
I will modify the README.md text slightly, before the new release.

I am not very sure how to use the pull request options. I pulled the pull request into a branch and then merged the files.

**Pull request closed without merging**